gstd/gstd_log.c

```diff
@@ -48,40 +48,72 @@ static FILE *_gstlog = NULL;
 static gchar *gstd_filename;
 static gchar *gst_filename;
 
-void
+gboolean
 gstd_log_init (const gchar * gstdfilename, const gchar * gstfilename)
 {
-  gint debug_color;
+  gboolean ret = TRUE;
+  gchar *err_filename;
 
   if (_gstdlog) {
-    return;
+    goto out;
   }
 
   gstd_filename =
       gstd_log_get_filename (gstdfilename, gstd_log_get_gstd_default ());
+
   _gstdlog = g_fopen (gstd_filename, "a+");
 
   if (!_gstdlog) {
-    g_printerr ("Unable to open Gstd log file: %s\n", g_strerror (errno));
-    return;
+    err_filename = gstd_filename;
+    goto error;
   }
 
   gst_filename =
       gstd_log_get_filename (gstfilename, gstd_log_get_gst_default ());
   _gstlog = g_fopen (gst_filename, "a+");
 
   if (!_gstlog) {
-    g_printerr ("Unable to open Gst log file: %s\n", g_strerror (errno));
-    return;
+    err_filename = gst_filename;
+    goto error;
+  }
+
+  /* Install our proxy handler */
+  gst_debug_add_log_function (gstd_log_proxy, NULL, NULL);
+  goto out;
```

Comment:
return here doesn't make sense to jump just to return

Reply:
Fixed